CONTRIBUTING.md

```diff
@@ -42,6 +42,25 @@ If you need to undo this for any reason, you can run:
 git update-index --no-assume-unchanged go.*
 ```
 
+Additionally there are some [known issues](https://github.com/DataDog/dd-trace-go/issues/911) caused by upstream modules not following semantic versioning. This means you're prone to see errors like this during local development:
```

Comment:
I think this overcomplicates the contributor guidelines unnecessarily and further underlines an issue that we didn't handle which will likely not affect 99% of contributors. Furthermore, the install instructions already illustrate how not to do `go get -v ./...`

This also encourages people to add even more commands (and tools, like `direnv`) such as GOFLAGS on top, only to account for one small potentially outdated integration.

I would vote against this and keep things simple. I don't completely understand the benefit of it:
* If you're working locally, and you're working on this contribution (grpc.v12), then you have the right dependencies and not getting these errors
* If you're working on anything else, you're also not getting this error

In both cases, you're not getting the error. So why this?